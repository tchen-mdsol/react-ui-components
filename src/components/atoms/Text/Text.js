import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const propTypes = {
  children: PropTypes.string,
  color: PropTypes.string
}
const defaultProps = {
  color: '#333'
}

const StyledText = styled.div`
  color: ${props => props.color};
`

const Text = ({
  children,
  color
}) => {
  return (
    <StyledText color={color}>
      {children}
    </StyledText>
  )
}

Text.propTypes = propTypes
Text.defaultProps = defaultProps

export default Text
