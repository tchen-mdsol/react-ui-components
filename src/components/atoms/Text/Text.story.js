import React from 'react'
import { storiesOf } from '@storybook/react'
import Text from '.'

const story = storiesOf('atoms/Text', module)
  .add('Torng-Lin Chen', () => {
    return (
      <Text color='#c00'>Torng-Lin Chen</Text>
    )
  })

export default story
