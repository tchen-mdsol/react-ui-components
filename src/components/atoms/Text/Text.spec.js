import React from 'react'
import renderer from 'react-test-renderer'
import Text from '.'

test('Text is expected', () => {
  const text = renderer.create(
    <Text />
  ).toJSON()
  expect(text).toMatchSnapshot()
})
