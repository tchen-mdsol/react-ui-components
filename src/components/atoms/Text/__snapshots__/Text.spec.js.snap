// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text is expected 1`] = `
<div
  className="sc-bdVaJa dsMjwO"
  color="#333"
/>
`;
